### DISCUSSION TITLE
Store JSON in field other than text eg tiddlername!!fieldname##indexname

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#5807] [Q&A]
Folks,

With the upcoming improvements to field handling, at least the naming, is this the correct time to ask the following?

Is there a way we can address indexes in a field other than text?
- One could imagine us addressing such values as tiddlername!!fieldname##indexname ?
- This would allow sophisticated settings to be contained inside each instance of a type of tiddler, and automatically transferred or exported with the tiddler.
- I understand a field cant have a content "type" set, but perhaps a convention such as fieldname (containing content) and a matching fieldname-type=application/json could be set allowing more sophisticated content in fields.
Your thoughts?
- One method could be storing multiple tiddlers, or versions of the current tiddler in the tiddler itself, a quite desirable feature even in a single file wiki. Thus undo and redo could be done even after moving a tiddler to another wiki.

Can this already be done with a trick I am not aware of?
Your thoughts otherwise.

Thanks
Tones

**Comment by pmario** [comment]
>I understand a field cant have a content "type" set, but perhaps a convention such as fieldname (containing content) and a matching fieldname-type=application/json could be set allowing more sophisticated content in fields.
Your thoughts?

You should read "How could system-fields look like with 5.2.x" #5805  ... Possible types section. .. There may be a possibility to add types to fields. Even to "user" fields, which would make it possible, that a `myColor` field would automatically create a color-picker

**Reply by pmario** [reply]
JSON was 1 of the types that I thought would be possible. ... BUT I'm not sure how it could be implemented.
